### PR TITLE
Python model generation

### PR DESCRIPTION
**Pull request by chamini2**
~~We have to stop parsing the whole dbt project before generating these and instead do a "lightweight" parsing to just get source-dirs. If not, dbt analyzes the SQL files and realizes that some SQL files do not exist (they will exist after generation).~~ DONE

Fixes FEA-55 and #301

### CODE REVIEW

File: src/fal/cli/dbt_model_generator.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            file.write(sql_contents)
+
+
+def _check_path_safe_to_write(sql_path: Path, py_path: Path):
+    if sql_path.exists():
+        with open(sql_path, "r") as file:
+            contents = file.read()
+            if "FAL_GENERATED" not in contents:
+                raise RuntimeError(
+                    f"File '{sql_path}' not generated by fal would be overwritten by generated model of '{py_path}'. Please delete or move."
```
Comment (mederka) [inline src/fal/cli/dbt_model_generator.py:49]: "Please rename or remove" ?

File: src/fal/cli/dbt_model_generator.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                )
+
+
+def _find_python_files(source_dirs: Union[List[str], List[Path]]) -> List[Path]:
+    return list(
+        flatten(
+            map(
```
Comment (mederka) [inline src/fal/cli/dbt_model_generator.py:56]: Indentation here looks strange

File: src/fal/cli/dbt_model_generator.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                lambda d: (p for p in Path(d).rglob("*.py") if p.is_file()),
```
Comment (mederka) [inline src/fal/cli/dbt_model_generator.py:57]: Should define this lambda as a separate function? We will also need to process .ipynb files.

File: src/fal/cli/dbt_model_generator.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+__deps__
+
+*/
+
+SELECT * FROM {{ target.schema }}.{{ model.name }}
+"""
+
+
+def generate_python_dbt_models(fal_dbt: FalDbt):
```
Comment (mederka) [inline src/fal/cli/dbt_model_generator.py:26]: Why don't we generate python dbt models before initiating `fal_dbt`? All we need is a project dir.

File: src/fal/cli/flow_runner.py [hunk 5]
```diff
@@ -23,6 +24,11 @@
 
 def fal_flow_run(parsed: argparse.Namespace):
     fal_dbt = create_fal_dbt(parsed)
+
+    if parsed.experimental_python_models:
+        telemetry.log_call("experimental_python_models")
+        generate_python_dbt_models(fal_dbt)
```
Comment (mederka) [inline src/fal/cli/flow_runner.py:30]: In the end we will have to clean these up, right?

**Review by mederka — COMMENTED** [review]
Very interesting implementation, @chamini2. I like the use of `ast`.

The biggest comment I have is why do we need to initiate `FalDbt` before generating SQL files? As long as we know the `project_dir` and can read `dbt_project.yml`, we should be able to find all the relevant python files.

File: src/fal/cli/dbt_model_generator.py (hunk 2 above)
Comment (chamini2) [reply src/fal/cli/dbt_model_generator.py:56]: lol, it's actually fine, 4 spaces each time

File: src/fal/cli/dbt_model_generator.py (hunk 3 above)
Comment (chamini2) [reply src/fal/cli/dbt_model_generator.py:57]: Not sure if we will process ipynb as models or just as hooks.

File: src/fal/cli/dbt_model_generator.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    '''
+    Simple regex analysis for docstring in top of the file. User can list dependencies one per line, but not multiline.
+    Example:
+
+    """
+    A Python model with some docstring introduction.
+
+    Dependencies:
+    - ref('model')
+    - source('some', 'table')
```
Comment (turbo1912) [inline src/fal/cli/dbt_model_generator.py:95]: do we need source dependencies here? Dont we need this just for dependency resolution? Does listing the source actually change anything?

**Review by turbo1912 — APPROVED** [review]

File: src/fal/cli/dbt_model_generator.py (hunk 6 above)
Comment (chamini2) [reply src/fal/cli/dbt_model_generator.py:95]: You are right, they probably don't do anything. I can just remove source for this.

**Pull request merged**